tests/unittest/_torch/attention/sparse/test_flashinfer_utils.py

```
@@ -0,0 +1,244 @@
+        fmha_libs=[],
+        is_mla_enable=True,
+        kv_cache_dtype="auto",
+        sparse_params=SimpleNamespace(algorithm="dsa"),
+    )
+
+    with (
+        patch(
+            "tensorrt_llm._torch.attention_backend.trtllm.get_sm_version",
+            return_value=120,
+        ),
+        pytest.raises(ValueError, match="requires kv_cache_config.dtype='fp8_ds_mla'"),
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)